src/KeyringController.ts

```diff
@@ -662,48 +861,36 @@ class KeyringController extends EventEmitter {
         });
       }
     } else {
+      if (typeof password !== 'string') {
+        throw new TypeError(
+          'KeyringControlle - password must be of type string',
+        );
+      }
+
       vault = await this.encryptor.decrypt(password, encryptedVault);
       this.password = password;
     }
 
-    await Promise.all(vault.map(this._restoreKeyring.bind(this)));
-    await this._updateMemStoreKeyrings();
+    await Promise.all(vault.map(this.#restoreKeyring.bind(this)));
+    await this.#updateMemStoreKeyrings();
     return this.keyrings;
   }
 
-  /**
-   * Restore Keyring
-   *
-   * Attempts to initialize a new keyring from the provided serialized payload.
-   * On success, updates the memStore keyrings and returns the resulting
-   * keyring instance.
-   *
-   * @param {object} serialized - The serialized keyring.
-   * @returns {Promise<Keyring>} The deserialized keyring.
-   */
-  async restoreKeyring(serialized) {
-    const keyring = await this._restoreKeyring(serialized);
-    if (keyring) {
-      await this._updateMemStoreKeyrings();
-    }
-    return keyring;
-  }
-
   /**
    * Restore Keyring Helper
    *
    * Attempts to initialize a new keyring from the provided serialized payload.
    * On success, returns the resulting keyring instance.
    *
-   * @param {object} serialized - The serialized keyring.
-   * @returns {Promise<Keyring|undefined>} The deserialized keyring or undefined if the keyring type is unsupported.
+   * @param serialized - The serialized keyring.
+   * @returns The deserialized keyring or undefined if the keyring type is unsupported.
    */
-  async _restoreKeyring(serialized) {
+  async #restoreKeyring(serialized: any): Promise<Keyring<State> | undefined> {
```

Comment:
Since `type` and `data` are being destructured out, should we give a type to `serialized`?

Reply:
Improved in [ce7d126](https://github.com/MetaMask/KeyringController/pull/202/commits/ce7d126375790bd07789a82f6c9609f0a2eb9813)